### PR TITLE
Add support for No new privs

### PR DESCRIPTION
**Pull request by rhatdan**
Also add new tests for other security checks.

### CODE REVIEW

File: libpod/container.go [hunk 1]
```diff
@@ -194,6 +194,8 @@ type ContainerConfig struct {
 	CreatedTime time.Time `json:"createdTime"`
 	// Cgroup parent of the container
 	CgroupParent string `json:"cgroupParent"`
+	// NoNewPrivs indicates whether the container will run with NoNewPrivs
+	NoNewPrivs bool `json:"noNewPrivs"`
```
Comment (mheon) [inline libpod/container.go]: Can you throw this a bit higher up in the ContainerConfig struct, with the other security-related options (SELinux labels and user)?

**Comment by mheon** [comment]
Needs DB changes to save NoNewPrivs in ContainerConfig

**Comment by rhatdan** [comment]
@mheon I don't think we need to save this, since it is part or the security opt?  If you do a podman inspect you will see the nonewprivs flag, just like you do in Docker.

I did notice that we need to save --privilege flag.

**Comment by mheon** [comment]
If it's in ContainerConfig it needs to be saved - we need to keep it around for Exec and Inspect

File: cmd/podman/create.go [hunk 2]
```diff
@@ -241,7 +242,7 @@ func parseSecurityOpt(config *createConfig, securityOpts []string) error {
 
 	for _, opt := range securityOpts {
 		if opt == "no-new-privileges" {
-			config.NoNewPrivileges = true
+			config.NoNewPrivs = true
```
Comment (TomSweeneyRedHat) [inline cmd/podman/create.go]: Does this need to be added to the createConfig at ~line 542 in parseCreateOpts?

**Comment by rhatdan** [comment]
@mheon I did not think of Exec, I have added it to the config.

File: cmd/podman/create.go (hunk 2 above)
Comment (rhatdan) [reply cmd/podman/create.go]: I believe it is in there.

**Comment by mheon** [comment]
LGTM, though we should hold off on merging until the queue clears

**Comment by rhatdan** [comment]
@bot retest this please

**Comment by mheon** [comment]
bot, retest this please

**Comment by rhatdan** [comment]
bot, retest this please

**Comment by rhatdan** [comment]
I think this is failing because it needs a new container-selinux in Fedora to pass.

**Comment by mheon** [comment]
@rhatdan I'm working on some DB changes to let @baude proceed with DNS work. I'm thinking about including the DB changes from here and #220 at the same time so we don't have 3 PRs out at once changing the DB. Any objections?

**Comment by rhatdan** [comment]
Nope since I am waiting for packages to be udpated, so I can pass tests. I will rebase once your stuff gets in.
Make sure you add privileged as well.

**Comment by rhatdan** [comment]
bot, retest this please

**Comment by rhatdan** [comment]
@bot, retest this please

**Comment by mheon** [comment]
bot, retest this please

**Comment by rhatdan** [comment]
@mheon @TomSweeneyRedHat @baude @umohnani8 PTAL

File: libpod/options.go [hunk 3]
```diff
@@ -280,6 +280,18 @@ func WithPrivileged(privileged bool) CtrCreateOption {
 	}
 }
 
+// WithNoNewPrivs sets the noNewPrivs flag in the container runtime
+func WithNoNewPrivs(noNewPrivs bool) CtrCreateOption {
```
Comment (mheon) [inline libpod/options.go]: Do we need this? It occurs to me that it's already in the spec, so we can get at the bool there

**Comment by mheon** [comment]
@rhatdan I think we can get rid of WithNoNewPrivs (and probably take it out of the DB too) - it's already in the spec, so we do store it

**Comment by umohnani8** [comment]
LGTM

File: libpod/options.go (hunk 3 above)
Comment (rhatdan) [reply libpod/options.go]: It needs to show up in `podman inspect` though.

Comment (mheon) [reply libpod/options.go]: Hm. We can probably fix that on the `libpod` side, but that can happen later. Now that we're backed by boltdb removing DB fields is no longer that big of a deal.

**Comment by mheon** [comment]
LGTM. We can get the changes I want in after release.

**Comment by mheon** [comment]
@rh-atomic-bot r+

**Comment by mheon** [comment]
Test failures on Fedora. Don't seem to be flakes?

**Comment by rhatdan** [comment]
These failures seem unrelated to the PR though?

**Comment by rhatdan** [comment]
bot, retest this please

**Comment by rhatdan** [comment]
Brent can you get this one over the finish line.

**Comment by mheon** [comment]
Closing in favor of #369 from @baude

**Pull request closed without merging**